off-chain/src/HTTP/Types.purs

```diff
@@ -0,0 +1,87 @@
+module Seath.HTTP.Types where
+
+import Contract.Prelude
+import Undefined
+
+import Aeson
+  ( class DecodeAeson
+  , class EncodeAeson
+  , decodeJsonString
+  , encodeAeson
+  , fromString
+  , isString
+  , stringifyAeson
+  , toString
+  )
+import Ctl.Internal.Test.UtxoDistribution (encodeDistribution)
+import Data.Bifunctor (bimap)
+import Data.String
+  ( Pattern(Pattern)
+  , Replacement(Replacement)
+  , replace
+  , replaceAll
+  )
+import Data.UUID (UUID)
+import Payload.Client.DecodeResponse (class DecodeResponse)
+import Payload.Client.EncodeBody (class EncodeBody)
+import Payload.ContentType (class HasContentType, json)
+import Payload.ResponseTypes (Json(..), Response(..), ResponseBody(..))
+import Payload.Server.DecodeBody (class DecodeBody)
+import Payload.Server.Response (class EncodeResponse, encodeResponse, ok)
+import Seath.Core.Types (UserAction)
+
+-- Include action
+-- ! acrh: newtype wrapper made to avoid having DecodeBody/EncodeBody
+-- ! instances in `Network` layer, as it should be decoupled from HTTP layer
+newtype IncludeRequest a = IncludeRequest (UserAction a)
+
+derive instance Newtype (IncludeRequest a) _
+derive newtype instance Show a => Show (IncludeRequest a)
+
+instance decIncludeReq ::
+  ( DecodeAeson a
+  ) =>
+  DecodeBody (IncludeRequest a) where
+  decodeBody s = bimap show IncludeRequest $ decodeJsonString s
+
+instance encIncludeReq ::
+  ( EncodeAeson a
+  ) =>
+  EncodeBody (IncludeRequest a) where
+  encodeBody (IncludeRequest action) = show $ encodeAeson action
+
+instance includeContentType :: HasContentType (IncludeRequest a) where
+  getContentType _ = json
+
+newtype UID = UID UUID
+
+instance showUID :: Show UID where
```

Comment:
I've tried to remove this `Show` instance and use `seathUUIDShow` instead, but it gave me `oversliced` UUID, e.g.
            `d2-be45-48db-b9a4-8cb7488d0e5`
vs
`50d21cd2-be45-48db-b9a4-8cb7488d0e56`

Reply:
yep, that's my fault, it assumes all UUIDs are built from `genUUID` and came with a prefix, so it just trips them.